src/notation/internal/notationinteraction.cpp

```diff
@@ -576,6 +576,29 @@ mu::async::Notification NotationInteraction::selectionChanged() const
     return m_selectionChanged;
 }
 
+bool NotationInteraction::isSelectionTypeFiltered(SelectionFilterType type) const
+{
+    return score()->selectionFilter().isFiltered(type);
+}
+
+void NotationInteraction::setSelectionTypeFiltered(SelectionFilterType type, bool filtered)
+{
+    score()->selectionFilter().setFiltered(type, filtered);
+    if (selection()->isRange()) {
+        score()->selection().updateSelectedElements();
+        notifyAboutSelectionChanged();
+    }
+}
+
+void NotationInteraction::setAllSelectionTypesFiltered(bool filtered)
```

Comment:
It seems that you can use the All type and setSelectionTypeFiltered instead of adding the separate method for this. Like this:

void NotationInteraction::setSelectionTypeFiltered(SelectionFilterType type, bool filtered)
{
if (type == All) {
    score()->selectionFilter().setAllFiltered(filtered);
} else {
    score()->selectionFilter().setFiltered(filtered);
}
...
}

Probably we can combine setAllFiltered and setFiltered too

Reply:
Thanks, good idea! Fixed.